### ISSUE TITLE
No HDMI Audio with AMDGPU-PRO on my XFX RX480

### ISSUE THREAD

**Issue by dvriv** [corngood/archlinux-amdgpu#44]
Any idea how to fix this? I have been reading some people his audio works while other don't.

```
Sink #0
	State: RUNNING
	Name: alsa_output.pci-0000_01_00.1.hdmi-stereo
	Description: HDA ATI HDMI Digital Stereo (HDMI)
	Driver: module-alsa-card.c
	Sample Specification: s16le 2ch 44100Hz
	Channel Map: front-left,front-right
	Owner Module: 6
	Mute: no
	Volume: front-left: 54384 /  83% / -4.86 dB,   front-right: 54384 /  83% / -4.86 dB
	        balance 0.00
	Base Volume: 65536 / 100% / 0.00 dB
	Monitor Source: alsa_output.pci-0000_01_00.1.hdmi-stereo.monitor
	Latency: 23194 usec, configured 23220 usec
	Flags: HARDWARE DECIBEL_VOLUME LATENCY SET_FORMATS 
	Properties:
		alsa.resolution_bits = "16"
		device.api = "alsa"
		device.class = "sound"
		alsa.class = "generic"
		alsa.subclass = "generic-mix"
		alsa.name = "HDMI 0"
		alsa.id = "HDMI 0"
		alsa.subdevice = "0"
		alsa.subdevice_name = "subdevice #0"
		alsa.device = "3"
		alsa.card = "1"
		alsa.card_name = "HDA ATI HDMI"
		alsa.long_card_name = "HDA ATI HDMI at 0xf7e60000 irq 31"
		alsa.driver_name = "snd_hda_intel"
		device.bus_path = "pci-0000:01:00.1"
		sysfs.path = "/devices/pci0000:00/0000:00:01.0/0000:01:00.1/sound/card1"
		device.bus = "pci"
		device.vendor.id = "1002"
		device.vendor.name = "Advanced Micro Devices, Inc. [AMD/ATI]"
		device.product.id = "aaf0"
		device.string = "hdmi:1"
		device.buffering.buffer_size = "352768"
		device.buffering.fragment_size = "176384"
		device.access_mode = "mmap+timer"
		device.profile.name = "hdmi-stereo"
		device.profile.description = "Digital Stereo (HDMI)"
		device.description = "HDA ATI HDMI Digital Stereo (HDMI)"
		alsa.mixer_name = "ATI R6xx HDMI"
		alsa.components = "HDA:1002aa01,00aa0100,00100700"
		module-udev-detect.discovered = "1"
		device.icon_name = "audio-card-pci"
	Ports:
		hdmi-output-0: HDMI / DisplayPort (priority: 5900, available)
	Active Port: hdmi-output-0
	Formats:
		pcm

Sink #1
	State: SUSPENDED
	Name: alsa_output.pci-0000_00_1b.0.analog-stereo
	Description: Built-in Audio Analog Stereo
	Driver: module-alsa-card.c
	Sample Specification: s16le 2ch 44100Hz
	Channel Map: front-left,front-right
	Owner Module: 7
	Mute: no
	Volume: front-left: 65536 / 100% / 0.00 dB,   front-right: 65536 / 100% / 0.00 dB
	        balance 0.00
	Base Volume: 65536 / 100% / 0.00 dB
	Monitor Source: alsa_output.pci-0000_00_1b.0.analog-stereo.monitor
	Latency: 0 usec, configured 0 usec
	Flags: HARDWARE HW_MUTE_CTRL HW_VOLUME_CTRL DECIBEL_VOLUME LATENCY 
	Properties:
		alsa.resolution_bits = "16"
		device.api = "alsa"
		device.class = "sound"
		alsa.class = "generic"
		alsa.subclass = "generic-mix"
		alsa.name = "ALC892 Analog"
		alsa.id = "ALC892 Analog"
		alsa.subdevice = "0"
		alsa.subdevice_name = "subdevice #0"
		alsa.device = "0"
		alsa.card = "0"
		alsa.card_name = "HDA Intel PCH"
		alsa.long_card_name = "HDA Intel PCH at 0xf7f30000 irq 30"
		alsa.driver_name = "snd_hda_intel"
		device.bus_path = "pci-0000:00:1b.0"
		sysfs.path = "/devices/pci0000:00/0000:00:1b.0/sound/card0"
		device.bus = "pci"
		device.vendor.id = "8086"
		device.vendor.name = "Intel Corporation"
		device.product.id = "1e20"
		device.product.name = "7 Series/C216 Chipset Family High Definition Audio Controller"
		device.form_factor = "internal"
		device.string = "front:0"
		device.buffering.buffer_size = "352800"
		device.buffering.fragment_size = "176400"
		device.access_mode = "mmap+timer"
		device.profile.name = "analog-stereo"
		device.profile.description = "Analog Stereo"
		device.description = "Built-in Audio Analog Stereo"
		alsa.mixer_name = "Realtek ALC892"
		alsa.components = "HDA:10ec0892,80862043,00100302"
		module-udev-detect.discovered = "1"
		device.icon_name = "audio-card-pci"
	Ports:
		analog-output-lineout: Line Out (priority: 9900, available)
		analog-output-speaker: Speakers (priority: 10000, not available)
		analog-output-headphones: Headphones (priority: 9000, not available)
	Active Port: analog-output-lineout
	Formats:
		pcm
```

**Comment by Gelmo** [comment]
You need the new DAL support, which is not yet in amdgpu-pro:

https://www.phoronix.com/scan.php?page=news_item&px=Ubuntu-AMDGPU-DC-Kernel

**Comment by corngood** [comment]
@Gelmo that's for the amdgpu driver in linux.  The one in amdgpu-pro has dal support.  However, we have this patch which changes the default state from enabled to disabled:

https://github.com/corngood/archlinux-amdgpu/blob/master/0007-disable-dal-by-default.patch

If you want to test with it enabled, you just need to boot with `amdgpu.dc=1`.  You can verify the state with `modinfo amdgpu`.

I had originally disabled it because of stability problems, but that was a few drivers ago.  If people generally have good results with it, we can enable it by default.  So let me know if it fixes anything for you, and if everything is still stable.

**Comment by Dav1605** [comment]
Not sure if it has something to do with the fact I used the AUR package instead of cloning it form here, but isn't working :(

```
title Arch Linux
linux /vmlinuz-linux
initrd /intel-ucode.img
initrd /initramfs-linux.img
options root=/dev/sda6 rw amdgpu.dc=1
```

Ant this is my modinfo amdgpu:
<script src="https://gist.github.com/Dav1605/0dc07d15d82ff613c139f0bcd3632e99.js"></script>

**Comment by corngood** [comment]
@dav1605

Are you sure you have the dkms driver installed correctly?  That looks like the amdgpu driver from linux.

**Comment by dvriv** [comment]
@corngood Oh thank you, it is working now!!! I'll let you know I see any bug or stability problems

**Comment by fjfalcon** [comment]
@corngood maybe you should close this issue?

**Comment by corngood** [comment]
Should be fixed in 17.10.  Please reopen if you still have problems.
